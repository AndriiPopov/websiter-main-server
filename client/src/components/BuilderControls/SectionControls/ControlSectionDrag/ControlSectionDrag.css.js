export const SectionDrag = {
    position: 'absolute',
    left: '-41px',
    top: '0px',
    height: '20px',
    width: '40px',
    background: '#ccc',
    cursor: 'move',
    textAlign: 'center',
    color: '#313131',
    lineHeight: '20px',
    fontSize: '11px'
};