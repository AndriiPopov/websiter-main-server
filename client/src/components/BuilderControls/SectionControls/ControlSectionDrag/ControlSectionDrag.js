import React from 'react';
import { connect } from 'react-redux';

import * as classes from './ControlSectionDrag.css.js';

export const ControlSectionDrag = props => {
    return !props.previewIsOn ? 
        <div style={props.chosenSection === props.id ? {...classes.SectionDrag, background: 'green'} : classes.SectionDrag }></div>
        : null;
};

const mapStateToProps = state => {
    return {
        chosenSection: state.builder.present.chosenSection,
        previewIsOn: state.builder.present.preview        
    };
};

export default connect(mapStateToProps)(ControlSectionDrag);