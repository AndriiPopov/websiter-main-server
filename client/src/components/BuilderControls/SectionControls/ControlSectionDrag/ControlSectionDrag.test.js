import React from 'react';
import renderer from 'react-test-renderer';

import { ControlSectionDrag } from './ControlSectionDrag';

describe('BoxControls', () => {
    it('should be defined', () => {
        expect(ControlSectionDrag).toBeDefined();
    });

    it('it should have green background when this section is chosen', () => {
        const tree = renderer.create(
            <ControlSectionDrag  id="element_879"
                chosenSection="element_879" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('it should not have green background when this section is not chosen', () => {
        const tree = renderer.create(
            <ControlSectionDrag  id="element_879"
                chosenSection="element_8579" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});