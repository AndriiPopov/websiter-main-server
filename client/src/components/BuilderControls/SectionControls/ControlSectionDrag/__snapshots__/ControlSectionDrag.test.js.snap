// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxControls it should have green background when this section is chosen 1`] = `
<div
  style={
    Object {
      "background": "green",
      "color": "#313131",
      "cursor": "move",
      "fontSize": "11px",
      "height": "20px",
      "left": "-41px",
      "lineHeight": "20px",
      "position": "absolute",
      "textAlign": "center",
      "top": "0px",
      "width": "40px",
    }
  }
/>
`;

exports[`BoxControls it should not have green background when this section is not chosen 1`] = `
<div
  style={
    Object {
      "background": "#ccc",
      "color": "#313131",
      "cursor": "move",
      "fontSize": "11px",
      "height": "20px",
      "left": "-41px",
      "lineHeight": "20px",
      "position": "absolute",
      "textAlign": "center",
      "top": "0px",
      "width": "40px",
    }
  }
/>
`;
